### ISSUE TITLE
MUSIC + LBT has repeatedly reported that the quest point is outside the light cone

### ISSUE THREAD

**Issue by K-JW** [JETSCAPE/JETSCAPE#41]
Hello, contributors,

Thanks for the code you contributed! 

When I run `MUSIC` + `LBT`, I often get a frustrating warning message that the query point is outside the light cone. 

> [Warning] Jetscape::FluidCellInfo Jetscape::EvolutionHistory::get_tz(Jetscape::real, Jetscape::real, Jetscape::real, Jetscape::real) const : the quest point is outside the light cone! t = 0, z = 0

I tried to avoid this problem by changing the configuration settings, but all the efforts I made did not affect. I want to ask what I should do to fix this problem. Hope someone can give me some guidance, thank you very much!

The **main program** is as follows:

```c++
#include <iostream>
#include "JetScape.h"
#include "JetEnergyLoss.h"
#include "JetEnergyLossManager.h"
#include "JetScapeWriterStream.h"
#include "LBT.h"
#include "MusicWrapper.h"
#include "CausalLiquefier.h"
#include "TrentoInitial.h"
#include "NullPreDynamics.h"
#include "PythiaGun.h"
#include "HadronizationManager.h"
#include "Hadronization.h"
#include "ColorlessHadronization.h"

using namespace std;
using namespace Jetscape;

int main(int argc, char** argv)
{
  JetScapeLogger::Instance()->SetInfo(true);
  JetScapeLogger::Instance()->SetDebug(false);
  JetScapeLogger::Instance()->SetRemark(false);
  JetScapeLogger::Instance()->SetVerboseLevel(0);

  auto jetscape = make_shared<JetScape>();
  jetscape->SetXMLMasterFileName("./jetscape_master.xml");
  jetscape->SetXMLUserFileName("./jetscape_user.xml");
  jetscape->SetId("primary");
  jetscape->SetReuseHydro (true);

  // Initial conditions and hydro
  auto trento = make_shared<TrentoInitial>();
  auto null_predynamics = make_shared<NullPreDynamics> ();
  auto pGun= make_shared<PythiaGun> ();
  auto hydro = make_shared<MpiMusic> ();
  jetscape->Add(trento);
  jetscape->Add(null_predynamics);
  jetscape->Add(pGun);
  jetscape->Add(hydro);

  // Energy loss
  auto jlossmanager = make_shared<JetEnergyLossManager> ();
  auto jloss = make_shared<JetEnergyLoss> ();
  auto myliquefier = make_shared<CausalLiquefier> ();
  jloss->add_a_liquefier(myliquefier);
  auto lbt = make_shared<LBT>();
  jloss->Add(lbt);
  jlossmanager->Add(jloss);  
  jetscape->Add(jlossmanager);

  auto writer= make_shared<JetScapeWriterAscii> ("test_out.dat");
  jetscape->Add(writer);

  jetscape->Init();
  jetscape->Exec();
  jetscape->Finish();

  return 0;
}
```

The **user profile** ("./jetscape_user.xml") is as follows:

```xml
<?xml version="1.0"?>

<jetscape>
  
  <!-- General settings -->
  <nEvents> 10 </nEvents>
  <setReuseHydro> true </setReuseHydro>
  <nReuseHydro> 10 </nReuseHydro>
  <JetScapeWriterAscii> on </JetScapeWriterAscii>
  
  <!-- Technical settings -->
  <debug> off </debug>
  <remark> off </remark>
  <vlevel> 0 </vlevel>
  <enableAutomaticTaskListDetermination> false </enableAutomaticTaskListDetermination>

  <!-- Inital State Module  -->
  <IS>
    <grid_max_x> 15 </grid_max_x>
    <grid_max_y> 15 </grid_max_y>
    <grid_max_z> 15 </grid_max_z>
    <grid_step_x> 0.3 </grid_step_x>
    <grid_step_y> 0.3 </grid_step_y>
    <grid_step_z> 0.3 </grid_step_z>
    
    <Trento>
    <PhysicsInputs  projectile='Pb'
            target='Pb'
            sqrts='2760'
            cross-section='6.4'
            normalization='13.9'>
    </PhysicsInputs>
    <CutInputs  centrality-low='50'
          centrality-high='60'>
    </CutInputs>
    <TransInputs  reduced-thickness='0.0'
            fluctuation="1.2"
            nucleon-width="0.9"
            nucleon-min-dist="1.2">
    </TransInputs>
    <LongiInputs  mean-coeff="1.0"
            std-coeff="3.0"
            skew-coeff="0.0"
            skew-type="1"
            jacobian="0.8">
    </LongiInputs>
    </Trento>
  </IS>
  
  <!-- Hard Process -->
  <Hard>
    <PythiaGun>
      <pTHatMin>50</pTHatMin>
      <pTHatMax>70</pTHatMax>
      <eCM>2760</eCM>
    </PythiaGun>
  </Hard>
  
  <!--Preequilibrium Dynamics Module -->
  <Preequilibrium>
    <NullPreDynamics> </NullPreDynamics>
  </Preequilibrium>
  
  <!--Eloss Modules -->
  <Eloss>
    <deltaT>0.1</deltaT>
    <formTime> -0.1</formTime>
    <maxT>10</maxT>
    <mutex>ON</mutex>
    <AddLiquefier> false </AddLiquefier>
	
	<Lbt>
      <name> Lbt </name>
      <Q0> 2.0 </Q0>
      <in_vac> 0 </in_vac>
      <only_leading> 0 </only_leading>
      <hydro_Tc> 0.16 </hydro_Tc>
      <alphas> 0.2 </alphas>
    </Lbt>
  </Eloss>
  
  <Hydro>
	<!-- MUSIC  -->
    <MUSIC>
      <name>MUSIC</name>
      <MUSIC_input_file>music_input</MUSIC_input_file>
      <output_evolution_to_file>1</output_evolution_to_file>
      <shear_viscosity_eta_over_s>0.08</shear_viscosity_eta_over_s>
      <freezeout_temperature>-1</freezeout_temperature>
      <Perform_CooperFrye_Feezeout>0</Perform_CooperFrye_Feezeout>
    </MUSIC>
  </Hydro>
  
</jetscape>
```

**Besides**, the `"./jetscape_master.xml"` and `music_input` files use JetScape's default files.

**Comment by jdmulligan** [comment]
Hi @shanshan-cao, Any update / guidance here?

**Comment by K-JW** [comment]
Excuse me, does anyone have some comments on this? Is this a bug? Is it because of my wrong setting? Thanks!

**Comment by amajumder** [comment]
Hi K-JW, 
Sorry that our response time is a little slow, given the upcoming conference. 
can I ask what is the physical process that you want to simulate. I see that you have no shower and are directly using LBT, you also have a Q0, and a mutex on, even though there is only one e-loss routine used. The Q0 and Mutex are usually meant for the case when one uses multiple e-loss modules, eg., MATTER + LBT. The default setting in Pythia within JETSCAPE is to set FSR off, is that what you wanted? 

My guess, without having run this yet, is that the lack of a final state shower routine like MATTER is causing this. Would you consider, adding a MATTER vacuum like shower between the Pythia and the LBT routine, just asking. 

If not then I would suggest that you set the formation time in E-loss to 0.1 (not -0.1).

**Comment by K-JW** [comment]
Hello @amajumder ,

Thank you very much for your help. As you said, the configuration file in the issue description does have some problems. 

When I submitted this issue, I realized these problems. In fact, I tried more combinations of settings, for example, I just changed the `<Brick bjorken_expansion_on="false" start_time="0.6"> </Brick>` in the `config/jetscapte_user.xml` (you provided) to `<MUSIC> </MUSIC>`, and then input the modified `config/jetscapte_user.xml` and unmodified `config/jetscapte_master.xml` into `runJetscape`, and the warning was still generated. 

Using other energy loss models did not get similar warnings.

I am very uneasy about taking up your time. Thank you again for your help.

**Comment by K-JW** [comment]
Hi @amajumder ,

Sorry, I didn't make myself clear. My motivation is so simple, I just want to use `Pythia8` to generate some parton-level events (including FSR), and then use LBT for energy loss simulation. As you said in [**arXiv:1903.07706**](https://arxiv.org/abs/1903.07706), your model supports this. But I couldn't get the right result. Thanks!

**Comment by amajumder** [comment]
@chanwook-park does your latest bug fix, address this issue?

**Comment by chanwook-park** [comment]
Sorry for late replay! I think this issue is not related to the bug fix that I made in the MARTINI side.

**Comment by jdmulligan** [comment]
Given the manual solution by @amajumder above, I will close this. I have opened a new issue (#84) to implement an automatic solution to this at the framework level, to be added in the (hopefully) near future. Thanks @K-JW for raising this.
